great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1257,14 +1256,16 @@ def _create_validator(self, *, result_format: ResultFormatUnion) -> V1Validator:
         )
 
     @overload
-    def compute_metrics(self, metrics: Metric[_MetricResultT]) -> _MetricResultT: ...
+    def compute_metrics(
+        self, metrics: Metric[_MetricResultT]
+    ) -> _MetricResultT | MetricErrorResult: ...
```

Comment:
A `MetricErrorResult` is a `_MetricResultT` since it's defined as:
```
class MetricErrorResult(MetricResult[MetricErrorResultValue]): ...
```

Reply:
It won't necessarily match the type var type though right?

Reply:
Oh, I see. You are right.